apis/v1alpha2/shared_types.go

```diff
@@ -167,25 +199,35 @@ type RouteStatusGatewayReference struct {
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=253
 	Controller string `json:"controller"`
+
+	// Conditions describes the status of the route with respect to the
+	// Gateway. The "Admitted" condition must always be specified by controllers
+	// to indicate whether the route has been admitted or rejected by the Gateway,
+	// and why. Note that the route's availability is also subject to the Gateway's
+	// own status conditions and listener status.
+	//
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=8
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
 // RouteStatus defines the observed state that is required across
 // all route types.
 type RouteStatus struct {
-	// Gateways is a list of Gateways that are associated with the route,
-	// and the status of the route with respect to each Gateway. When a
-	// Gateway selects this route, the controller that manages the Gateway
-	// must add an entry to this list when the controller first sees the
-	// route and should update the entry as appropriate when the route is
-	// modified.
-	//
-	// A maximum of 100 Gateways will be represented in this list. If this list
-	// is full, there may be additional Gateways using this Route that are not
-	// included in the list. An empty list means the route has not been admitted
-	// by any Gateway.
-	//
-	// +kubebuilder:validation:MaxItems=100
-	Gateways []RouteGatewayStatus `json:"gateways"`
+	// Parents is a list of parent resources (usually Gateways) that are
+	// associated with the route, and the status of the route with respect to
+	// each parent. When this route attaches to a parent, the controller that
+	// manages the parent must add an entry to this list when the controller
+	// first sees the route and should update the entry as appropriate when the
+	// route is modified.
+	//
+	// A maximum of 32 Gateways will be represented in this list. An empty list
+	// means the route has not been admitted by any Gateway.
```

Comment:
If a route is attached but rejected, it should appear here, right?  
```suggestion
	// means the route has not been attached to any Gateway.
```

Reply:
Updated, thanks!